lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -183,27 +211,50 @@ def self.get_token
 
       def self.validate_app!(app_id)
         token = get_token
-
-        connection = Faraday.new(url: BASE_URL) do |conn|
-          conn.response(:json, parser_options: { symbolize_names: true })
-          conn.response(:raise_error) # raise_error middleware will run before the json middleware
-          conn.adapter(Faraday.default_adapter)
-        end
         begin
-          response = connection.get("#{PATH}#{app_id}") do |request|
+          response = CONNECTION.get("#{PATH}#{app_id}") do |request|
             request.headers["Authorization"] = "Bearer " + token
           end
         rescue Faraday::ResourceNotFound => error
           UI.user_error!("App Distribution could not find your app #{app_id}. Make sure to onboard your app by pressing the \"Get started\" button on the App Distribution page in the Firebase console: https://console.firebase.google.com/project/_/appdistribution")
         rescue => error
           UI.crash!("Failed to fetch app information: #{error.message}")
         end
-
         contact_email = response.body[:contactEmail]
 
         if contact_email.strip.empty?
           UI.user_error!("We could not find a contact email for app #{app_id}. Please visit App Distribution within the Firebase Console to set one up.")
         end
+        return "projects/#{response.body[:projectNumber]}/apps/#{response.body[:appId]}/releases/-/binaries/"
+      end
+
+      def self.upload_binary!(app_id, binary_path)
+        token = get_token
```

Comment:
Since you're using get_token in a number of places I think the "lazy" approach I mentioned above would be good for this method as well, that way you're not making a fresh request (and fetching a new access token) every time.

Reply:
+1. Remove the "bang" (`!`) based on my previous comment.